workflow-config.yml

```diff
@@ -5,13 +5,15 @@ workflows:
     handler: argo-workflow
```

Comment:
I thought we discussed this, but reviewing the initial Pr for running a workflow I didn't see this comment, so maybe we just talked about it and I didn't write it down, so we both overlooked it.

We really need to have both a `handler` property and a `type` property. The `handler` is effectively something like the "queue" into which the action is inserted, or perhaps the "instance name" of the thing doing the action processing. A `swoop-conductor` instance will be configured to listen to a specific set of `handler`s, and will dispatch actions to those `handler`s.

But the `handler` name is independent of the actual workflow type, i.e., whether it is an argo workflow or a cirrus workflow, and therefore `handler` should not be used as the key in the discriminate union, as it is disconnected from the workflow type.

So we need to update the configs. Maybe set `handler` like `argo-handler` and `cirrus-handler`, and `type` like `argo-workflow` and `cirrus-workflow`. Then the `BaseWorkflow` model gets a `handler` string property, and the union needs to use `type` instead of `handler`.